File: tests/fixtures/patterns-positive/composite/mutB.ts
// POSITIVO (mutación B: recorrido de árbol repetido, sin abstracción compartida).
class ShapeGroup {
  private children: Shape[] = [];
  add(child: Shape): void {
    this.children.push(child);
  }
  render(): void {
    this.children.forEach((c) => c.render());
  }
}
class WidgetGroup {
  private children: Widget[] = [];
  add(child: Widget): void {
    this.children.push(child);
  }
  render(): void {
    this.children.forEach((c) => c.render());
  }
}

File: tests/fixtures/patterns-positive/composite/mutB.py
# POSITIVO (mutación B: recorrido de árbol repetido, sin abstracción compartida).
class ShapeGroup:
    def __init__(self):
        self.children = []

    def add(self, child):
        self.children.append(child)

    def render(self):
        for c in self.children:
            c.render()


class WidgetGroup:
    def __init__(self):
        self.children = []

    def add(self, child):
        self.children.append(child)

    def render(self):
        for c in self.children:
            c.render()
